### PR TITLE
@damassi => Initial component + story for the follow artist popover w/ artist suggestions

### PR DESCRIPTION
**Pull request by mzikherman**
cc @iskounen this is what we were discussing pairing on Monday.

This is just an initial, unstyled attempt at the suggested artist popover.

bp

Basically this starts with rendering a list of suggested artists, based on a starting artist. Then, as you click on an artist, you follow them, and that row is replaced with a new artist suggestion based on the one you clicked.

### CODE REVIEW

File: src/Styleguide/Components/FollowArtistPopover/index.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+              ...FollowArtistPopoverRow_artist
+            }
+          }
+        }
+      }
+    }
+  `
+)
+
+export const FollowArtistPopoverQueryRenderer = ({
```
Comment (mzikherman) [inline src/Styleguide/Components/FollowArtistPopover/index.tsx]: It's very possible this won't have its own query renderer when being used along with the Follow button on an existing/new page. But for now (+ storybooks), we'll include it.

File: src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+interface State {
+  swappedArtist: FollowArtistPopoverRow_artist
+}
+
+class FollowArtistPopoverRow extends React.Component<Props, State> {
+  state = {
+    swappedArtist: null,
+  }
+  handleClick(artistID: string, relayID: string) {
```
Comment (damassi) [inline src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx]: line break

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit f71872beb31c07ad7f713c3d59e70b673b04782a

https://deploy-preview-1489--artsy-reaction.netlify.com

File: src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { artist_id: artistID, unfollow: false },
+        },
+        updater: (store: RecordSourceSelectorProxy, data: SelectorData) => {
+          const newEdge = data.followArtist.artist.related.suggested.edges[0]
+          this.setState({ swappedArtist: newEdge.node })
```
Comment (mzikherman) [inline src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx]: This component renders a row for a given artist, which can either be the artist it started out with, or one that gets swapped in as you click + follow (which is set in state).

At first, I was trying to update the Relay store to remove the clicked on artist and replace it with the new one, but that seems like overkill/unnecessary when this `Row` component can just use its own state for that.

File: src/Styleguide/Components/FollowArtistPopover/index.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import styled from "styled-components"
+import { FollowArtistPopoverRowFragmentContainer as FollowArtistPopoverRow } from "./FollowArtistPopoverRow"
+
+const Container = styled.div``
+
+interface Props extends ContextProps {
+  suggested: FollowArtistPopover_suggested
+}
+
+class FollowArtistPopover extends React.Component<Props, null> {
```
Comment (damassi) [inline src/Styleguide/Components/FollowArtistPopover/index.tsx]: Can you convert this to a SFC? As we're preparing for the new React we should only use classes when state or tracking is required.

File: src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      __id
+      name
+      image {
+        cropped(width: 100, height: 100) {
+          url
+        }
+      }
+    }
+  `
+)
```
Comment (damassi) [inline src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx]: Can you stub out some tests? Thinking that we should get in the habit of writing tests alongside new code. (Totally cool to return to this tho)

File: src/Styleguide/Components/FollowArtistPopover/index.tsx (hunk 4 above)
Comment (mzikherman) [reply src/Styleguide/Components/FollowArtistPopover/index.tsx]: Yep, totally! 👍

File: src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx (hunk 5 above)
Comment (mzikherman) [reply src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx]: 100%, will do.

**Comment by mzikherman** [comment]
> Couple small comments -- merge when ready

Cool, I addressed those, will add tests and merge.

File: src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          mutation FollowArtistPopoverRowMutation($input: FollowArtistInput!) {
+            followArtist(input: $input) {
+              artist {
+                __id
+                related {
+                  suggested(first: 1, exclude_followed_artists: true) {
+                    edges {
+                      node {
+                        __id
+                        ...FollowArtistPopoverRow_artist @relay(mask: false)
```
[the remark this answers is not in the document]
Comment (alloy) [reply src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx]: Yup yup, this is a good usage of it 👍

**Comment by mzikherman** [comment]
K, spec added!

File: src/Styleguide/Components/FollowArtistPopover/__tests__/FollowArtistPopover.test.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            }}
+          />
+        </MockBoot>
+      </RelayStubProvider>
+    )
+  }
+
+  describe("general behavior", () => {
+    beforeAll(async () => {
+      wrapper = await getWrapper()
```
Comment (damassi) [inline src/Styleguide/Components/FollowArtistPopover/__tests__/FollowArtistPopover.test.tsx]: @mzikherman - I imagine this is from previous impl but no need to async / await with RelayStubProvider -- all sync.

Comment (mzikherman) [reply src/Styleguide/Components/FollowArtistPopover/__tests__/FollowArtistPopover.test.tsx]: Ah, thanks for the heads up.

**Pull request merged**